 <img src="https://c02.purpledshub.com/uploads/sites/48/2024/05/solar-flare-10-may-2024.jpg?webp=1&w=1200" width="400">



> One intersting aspect of the Sun is its sunspots. Sunspots are areas where the magnetic field is about 2,500 times stronger than Earth's, much higher than anywhere else on the Sun. Because of the strong magnetic field, the magnetic pressure increases while the surrounding atmospheric pressure decreases. This in turn lowers the temperature relative to its surroundings because the conencentrated magnetic field inhibits the flow of hot, new gas from the Sun's interior to the surface.


>Sumspots tend to occur in pairs that have magnetic fields pointing in opposite directions. A typical spot consists of the dark region called the umbra, surrounded by a lighter region known as the peumbra. The sunspots appear relatively dark because the surrounding surface of the sun (the photospher) is about 10,000 degrees F., while the umbra is about 6,300 degrees F. Sunspots are quite large as an average size about the same size as the Earth.


>Sunspots increase and decrease, then again increases and decreases through an average cycle of 11 years. Solar cycle 1 was the first solar cycle since extensive recording of solar suspot activity began in 1755 and we have experienced a total of 24 full solar cycle since then. Currently, we are in 25th solar cycle, the solar cycle 25 began in December 2019 and is expected to last till more 11 years that is about 2030.


<div align="center">
<img src="
https://www.cosmos.esa.int/documents/519784/530468/Sunspot+case+web+image/e505728f-9b2a-4192-8ba3-3f6243a59b7b?t=1475670734908" width="400">
</div>






In [2]:
from google.colab import files
files.upload()

Saving kaggle.json to kaggle.json


{'kaggle.json': b'{"username":"mahoorshams","key":"2b5ec3a9e9700362b646e3b095c12b8d"}'}

In [3]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("robervalt/sunspots")

print("Path to dataset files:", path)

100%|██████████| 22.4k/22.4k [00:00<00:00, 18.5MB/s]

Extracting files...
Path to dataset files: /root/.cache/kagglehub/datasets/robervalt/sunspots/versions/3


In [5]:
import os

path = "/root/.cache/kagglehub/datasets/robervalt/sunspots/versions/3"

for root, dirs, files in os.walk(path):
    for file in files:
        print(os.path.join(root, file))

/root/.cache/kagglehub/datasets/robervalt/sunspots/versions/3/Sunspots.csv


In [7]:
'''Visualization'''
import matplotlib.pyplot as plt
import seaborn as sns

'''Data Handeling and Linear Algebra'''
import pandas as pd
import numpy as np

'''Manipulating Data and Model Building'''
import tensorflow as tf

'''Ignore Warnings'''
import warnings
warnings.filterwarnings('ignore')

'''Change the plot style'''
import matplotlib.pyplot as plt
plt.style.use('fivethirtyeight')



In [9]:
# Take a look at the data
data = pd.read_csv("/root/.cache/kagglehub/datasets/robervalt/sunspots/versions/3/Sunspots.csv")
data.head()

Unnamed: 0        Date  Monthly Mean Total Sunspot Number
0           0  1749-01-31                               96.7
1           1  1749-02-28                              104.3
2           2  1749-03-31                              116.7
3           3  1749-04-30                               92.8
4           4  1749-05-31                              141.7

In [10]:
# Check the last year in the data
data.tail()

Unnamed: 0        Date  Monthly Mean Total Sunspot Number
3260        3260  2020-09-30                                0.6
3261        3261  2020-10-31                               14.4
3262        3262  2020-11-30                               34.0
3263        3263  2020-12-31                               21.8
3264        3264  2021-01-31                               10.4

In [11]:
# General information about the data
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3265 entries, 0 to 3264
Data columns (total 3 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Unnamed: 0                         3265 non-null   int64  
 1   Date                               3265 non-null   object 
 2   Monthly Mean Total Sunspot Number  3265 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 76.7+ KB


---
*   Total of 3265 months of data starting from january 1749 till january 2021
*   No null values found in the data
*   We are all set to extract our sunspots data from this dataframe and visulize it
---



In [12]:
sunspots = data.iloc[:, -1]
sunspots.values

array([ 96.7, 104.3, 116.7, ...,  34. ,  21.8,  10.4])